scripts/build_egs_potentials.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+"""Build EGS potentials for different depths and temperatures. Upper limits available
+heat content by overlaying subsurface potential with geospatial demand data."""
+
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+import os
+
+import rasterio
+import numpy as np
+import xarray as xr
+import pandas as pd
+from tqdm import tqdm
+import geopandas as gpd
+from rasterio.transform import xy
+from shapely.geometry import Point
+
+from _helpers import configure_logging
+
+
+def get_raster_file(tif_file):
+
+    with rasterio.open(tif_file) as src:
+
+        band1 = src.read(1)
+        transform = src.transform
+        rows, cols = band1.shape
+
+        row_indices, col_indices = np.meshgrid(np.arange(rows), np.arange(cols), indexing='ij')
```

Comment:
As a little technical comment, `np.meshgrid` has lead to memory issues when I have tested it for dummy data. Not sure how big is the concern in case of real data but generally `meshgrid` is known to be memory-intensive while other options can be used to alleviate that like in [this](https://stackoverflow.com/q/2460627/8465924) and [this](https://stackoverflow.com/q/58992019/8465924) SO discussions.